### DISCUSSION TITLE
Debug Logging: Web Follower

### DISCUSSION THREAD

**Discussion by schorschl** [NightscoutFoundation/xDrip#2702] [Q&A] [answered]
Hey everyone,
I hope this is the right place to ask.

I am using "Web Follower" as Data Source in xDrip+ (2022.12.24) with a Libre Link Up Account which gets the data from my Libre 3 (EU).
The poll sometimes does not work for 10-20minutes and I don't get any readings. I do not know if it is the Libre Service blocking or a network issue or maybe even the xDrip+ service failing.
I cannot find any error message on the "Web Follower Status" page or log at all, I would like to enable debug logging for the WebFollow Data Source. But I cannot find any list or description which "tag" would be correct and valid for this service to log.
Is there a list of tags or how can I debug the WebFollow service? Or any other tip how I can debug this?

Thanks in advance.

PS: 
* I followed https://xdrip.readthedocs.io/en/latest/install/webfollower/#prerequisites
* Sensor is started
* I have accepted the "privacy policy"/"terms and conditions" in the Link Up App: https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Comment by schorschl** [comment]
I have just tried and set "WebFollower:v" as a tag which apparently works.

Is the WebFollower service designed to request readings each 10 minutes instead of 5 minutes?

**Reply by Navid200** [reply]
> Is the WebFollower service designed to request readings each 10 minutes instead of 5 minutes?  
  
I don't think so.  Otherwise, I would have heard this before.

**Reply by schorschl** [reply]
At the moment it seems like the web connections work and get the data each 5 minutes. 
Sometimes I see the error "Failed to get any response" but this seems to be related to IPv6. 
I need some more time to analyze this and figure out how I can get the logs from my mobile. I am totally new to xDrip+ and the option to save is greyed out. ;-)

**Reply by schorschl** [reply]
What I have found out so far:

xDrip+ and Web Follower works as designed and get's the results each 5mins from LLU.

Problems occur most of the time because the FSL3 app did not upload data or LLU did not include new data in the response for 5-20 minutes. The web response includes all historical values but the latest 5-20 minutes (1-4 values) are missing.
As soon as those missing values are transmitted, xDrip+ shows those in the graph as well.

If I find the time I will try to find out if the FSL3 app does not upload the latest values or if the LLU service did not process it and failed to present it.

**Comment by schorschl** [comment, marked as answer]
If anybody wants to debug the WebFollower service even closer, set "WebFollower:v" as a log tag.

This way I could find out that the problem at that time was based on the Libre Link Up webservice which delivered old data and xDrip+ was working as expected.

**Reply by Navid200** [reply]
Thanks
